ompi/mca/coll/spacc/coll_spacc_exscan.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+#include "coll_spacc.h"
+
+#include "mpi.h"
+#include "ompi/constants.h"
+#include "opal/util/bit_ops.h"
+#include "ompi/datatype/ompi_datatype.h"
+#include "ompi/communicator/communicator.h"
+#include "ompi/mca/coll/coll.h"
+#include "ompi/mca/coll/base/coll_base_functions.h"
+#include "ompi/mca/coll/base/coll_tags.h"
+#include "ompi/mca/coll/base/coll_base_util.h"
+#include "ompi/mca/pml/pml.h"
+#include "ompi/op/op.h"
+
+/*
+ * mca_coll_spacc_exscan_intra_recursivedoubling
+ *
+ * Function:  Recursive doubling algorithm for exclusive scan.
+ * Accepts:   Same as MPI_Exscan
+ * Returns:   MPI_SUCCESS or error code
+ *
+ * Description:  Implements recursive doubling algorithm for MPI_Exscan.
+ *               The algorithm preserves order of operations so it can
+ *               be used both by commutative and non-commutative operations.
+ *
+ * Example for 5 processes and commutative operation MPI_SUM:
+ * Process:  0                 1             2              3               4
+ *  rbuf:    -                 -             -              -               -
+ *  psend:  [0]               [1]           [2]            [3]             [4]
+ *
+ * Step 1:
+ *  rbuf:    -                [0]            -             [2]              -
+ *  psend:  [1+0]             [0+1]         [3+2]          [2+3]           [4]
+ *
+ * Step 2:
+ *  rbuf:    -                [0]            [1+0]          [(0+1)+2]       -
+ *  psend:  [(3+2)+(1+0)]     [(2+3)+(0+1)]  [(1+0)+(3+2)]  [(1+0)+(2+3)]  [4]
+ *
+ * Step 3:
+ *  rbuf     -                [0]            [1+0]          [(0+1)+2]      [(3+2)+(1+0)]
+ *  psend:  [4+((3+2)+(1+0))]                                              [((3+2)+(1+0))+4]
+ *
+ * Time complexity (worst case): \ceil(\log_2(p))(2\alpha + 2m\beta + 2m\gamma)
+ * Memory requirements (per process): 2 * count * typesize = O(count)
+ * Limitations: intra-communicators only
+ */
+int mca_coll_spacc_exscan_intra_recursivedoubling(
```

Comment:
`mca_coll_spacc_exscan_intra_recursivedoubling()` should simply invoke `mca_coll_base_exscan_intra_recursive_doubling()`